src/term/mod.rs

```diff
@@ -363,6 +371,30 @@ impl Term {
       }
     }
   }
+
+  pub fn num_match(scrutinee: Term, zero_term: Term, succ_label: Option<Name>, succ_term: Term) -> Self {
+    let zero = (RulePat::Num(MatchNum::Zero), zero_term);
+    let succ = (RulePat::Num(MatchNum::Succ(succ_label)), succ_term);
+    Self::Match { scrutinee: Box::new(scrutinee), arms: vec![zero, succ] }
+  }
+}
+
+pub fn native_match(arms: Vec<(RulePat, Term)>) -> Option<(Term, Term)> {
+  use MatchNum::*;
+
+  if !arms.len() == 2 {
```

Comment:
You can convert this into a slice match https://doc.rust-lang.org/reference/patterns.html#slice-patterns